src/Sarif.Driver/OrderedFileSpecifier.cs

```diff
@@ -87,27 +87,24 @@ private IEnumerable<IEnumeratedArtifact> EnumeratedArtifacts()
 
             Task directoryEnumerationTask;
 
-            if (this.recurse)
+            directoryEnumerationTask = Task.Run(() =>
```

Comment:
The issue and fix is simple, in the recurse false code path we missed to use `Task.Run` to run the file enumeration operation asynchronously on the separate thread.
Now we merged in a same code block so both 
recurse true and false will have it.